File: codeforces/66.soft_drinking/soft_drinking.py
# Author: José Rodolfo (jric2002)
n, k, l, c, d, p, nl, np = list(map(int, input().split(" ")))
r = [((k * l) / nl), c * d, p / np]
number_toasts = int(min(r) / n)
print(number_toasts)

File: codeforces/low_difficulty/800_points/66.soft_drinking_[151A]/soft_drinking.cpp
#include <iostream>
#include <vector>
#include <algorithm>
/* Author: José Rodolfo */
using namespace std;
int main() {
  unsigned short int n, k, l, c, d, p, nl, np;
  vector<unsigned short int> r;
  unsigned short int number_toasts;
  cin >> n >> k >> l >> c >> d >> p >> nl >> np;
  r.push_back((k * l) / nl);
  r.push_back(c * d);
  r.push_back(p / np);
  vector<unsigned short int>::iterator aux = min_element(r.begin(), r.end());
  number_toasts = *aux / n;
  cout << number_toasts << endl;
  return 0;
}